subcommand/mesh-init/command.go

```diff
@@ -104,99 +92,58 @@ func (c *Command) realRun() error {
 			return err
 		}
 
-		// Temporary workaround so that unit tests relying on the
-		// previous version of the client (that talks through agents)
-		// can pass.
-		// TODO: Remove this after getting rid of the older version of the client
-		cfg.TokenFile = tokenFile
-
 		c.log.Info("wrote ACL token to shared volume", "token-file", tokenFile)
 	}
 
-	// Client config for the V2 client that talks directly to the
-	// server agent
-	consulClientCfg := c.config.ClientConfig()
-	consulClientCfg.Address = fmt.Sprintf("%s:%d", state.Address.IP.String(), c.config.ConsulServers.HTTPPort)
+	// Client config for the client that talks directly to the server agent
+	cfg := c.config.ClientConfig()
+	cfg.Address = fmt.Sprintf("%s:%d", state.Address.IP.String(), c.config.ConsulServers.HTTPPort)
```

Comment:
minor: consider using https://pkg.go.dev/net#JoinHostPort (in case of an IPv6 address it will bracket the address as-needed)